repos/fdbt-site/tests/pages/api/managePassengerGroup.test.ts

```diff
@@ -0,0 +1,265 @@
+                ['Regular infant-age-range-max']: '3',
+                ['Regular infant-proof-docs']: [],
+                passengerGroupName: 'My duplicate group',
+            },
+            uuid: {},
+            mockWriteHeadFn: writeHeadMock,
+        });
+        await managePassengerGroup(req, res);
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: '/managePassengerGroup',
+        });
+        expect(updateSessionAttributeSpy).toBeCalledWith(req, GS_PASSENGER_GROUP_ATTRIBUTE, {
+            errors: [
+                {
+                    errorMessage: 'There is already a group with this name. Choose another',
+                    id: 'passenger-group-name',
+                    userInput: 'My duplicate group',
+                },
+            ],
+            inputs: {
+                companions: [
+                    {
+                        ageRangeMax: '',
+                        ageRangeMin: '',
+                        maxNumber: '3',
+                        minNumber: '2',
+                        name: 'Regular Senior',
+                        passengerType: 'senior',
+                        proofDocuments: [],
+                    },
+                    {
+                        ageRangeMax: '3',
+                        ageRangeMin: '1',
+                        maxNumber: '2',
+                        minNumber: '1',
+                        name: 'Regular infant',
+                        passengerType: 'infant',
+                        proofDocuments: [],
+                    },
+                ],
+                maxGroupSize: '7',
+                name: 'My duplicate group',
+            },
+        });
+    });
+
+    it('should return 302 redirect to /viewPassengerTypes when there have been correct user inputs and the group is saved to the db', async () => {
+        const writeHeadMock = jest.fn();
+        getPassengerTypeByNameAndNocCodeSpy.mockResolvedValueOnce(undefined).mockResolvedValueOnce(undefined);
+        const { req, res } = getMockRequestAndResponse({
+            cookieValues: {},
+            body: {
+                maxGroupSize: '7',
+                passengerTypes: ['Regular Senior', 'Regular infant'],
+                ['minimumPassengersRegular Senior']: '2',
+                ['maximumPassengersRegular Senior']: '3',
+                ['Regular Senior-type']: 'senior',
+                ['Regular Senior-age-range-min']: '',
+                ['Regular Senior-age-range-max']: '',
+                ['Regular Senior-proof-docs']: [],
+                ['minimumPassengersRegular infant']: '1',
+                ['maximumPassengersRegular infant']: '2',
+                ['Regular infant-type']: 'infant',
+                ['Regular infant-age-range-min']: '1',
+                ['Regular infant-age-range-max']: '3',
+                ['Regular infant-proof-docs']: [],
+                passengerGroupName: 'My duplicate group',
```

Comment:
Not a big deal, but should we name this something else as it isn't a duplicate?

Reply:
yeah sure! good catch